internal/authtoken/query.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package authtoken
+
+const (
+	estimateCountAuthTokens = `
+select reltuples::bigint as estimate from pg_class where oid in ('auth_token'::regclass)
+`
+	listAuthTokensTemplate = `
+with auth_accounts as (
+    select public_id,
+           auth_method_id,
+           scope_id,
+           iam_user_id,
+           iam_user_scope_id
+      from auth_account
+     where scope_id in %s
+),
+final as (
+    select at.public_id,
+           at.auth_account_id,
+           aa.auth_method_id,
+           aa.scope_id,
+           aa.iam_user_id,
+           aa.iam_user_scope_id,
+           at.create_time,
+           at.update_time,
+           at.approximate_last_access_time,
+           at.expiration_time,
+           at.status 
+      from auth_token at
+      join auth_accounts aa on aa.public_id = at.auth_account_id
+  order by at.create_time desc, at.public_id asc
+     limit %d
```

Comment:
These two should be in the first CTE to make this efficient, I think. This might need to be something like 

```sql
with auth_tokens as (
  select public_id
   from auth_token
  order by at.create_time desc, at.public_id asc
  limit %d
),
auth_accounts as (
...
from auth_accounts
where %s -- Scope restriction is constructed
),
final as (
...
from auth_tokens at
join auth_accounts aa
)
select *
  from final
order by ...
```
That way, the very root of the query gets the limit applied efficiently using the create_time, public_id index.